src/main/java/algorithms/curated170/medium/binarytreeverticalordertraversal/BinaryTreeVerticalOrderTraversalBFS.java

```diff
@@ -0,0 +1,91 @@
+package algorithms.curated170.medium.braceexpansion;
+
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Deque;
+import java.util.List;
+import java.util.TreeMap;
+
+import algorithms.datastructures.TreeNode;
+
+public class BinaryTreeVerticalOrderTraversalBFS {
+
+  int nodeCount;
+  public List<List<Integer>> verticalOrder(TreeNode root) {
+    if (root == null) {
+      return Collections.emptyList();
+    }
+    
+    nodeCount = 0;
+    int[] range = new int[] {0, 0};
+    getRange(root, range, 0);
+
+    List<List<Integer>> columnsAtXPos = new ArrayList<>(range[1]-range[0] + 2);
+
+    for (int i = range[0]; i <= range[1]; i++) {
+        columnsAtXPos.add(new ArrayList<Integer>());
+    }
+    
+    Deque<NodePosPair> q = new ArrayDeque<>();
+    q.offer(new NodePosPair(root, -1*range[0]));
+
+    while (!q.isEmpty()) {
+      NodePosPair npp = q.poll();
+      npp.placeIntoMap(columnsAtXPos);
+      npp.queueChildren(q);
+    }
+    
+    
+    return columnsAtXPos;
+}
+
+public void getRange(TreeNode root, int[] range, int col) {
+    if (root == null) {
+        return;
+    }
+
+    nodeCount++;
+    range[0] = Math.min(range[0], col);
+    range[1] = Math.max(range[1], col);
+    
+    getRange(root.left, range, col - 1);
+    getRange(root.right, range, col + 1);
+}
+
+  private class NodePosPair {
```

Comment:
We can define NodePosPair in an external class file to use it in multiple classes.

Reply:
In the two BFS solutions, the implementation is not the same. So, I think it's not worth bothering as we won't use  it again and it would create problems while testing.

Reply:
Yeah, of course. No problem.